adl/language/parser.ts

```diff
@@ -0,0 +1,603 @@
+import { Kind, Scanner } from './scanner';
+
+export function parse(code: string) {
+  const s = new Scanner(code);
+  nextToken();
+  return parseADLScript();
+
+  function parseADLScript(): ADLScriptNode {
+    const script: ADLScriptNode = {
+      kind: SyntaxKind.ADLScript,
+      statements: [],
+      pos: 0,
+      end: 0
+    };
+
+    while (!s.eof) {
+      script.statements.push(parseStatement());
+    }
+
+    script.end = s.offset;
+    return script;
+  }
+
+  function parseStatement(): Statement {
+    let decorators = [];
+
+    // eslint-disable-next-line
```

Comment:
```suggestion
    // eslint-disable-next-line  no-constant-condition
```

I think it's nice to see what eslint was reporting when we suppress.

Also, in this case, I'd be in favor of reconfiguring the rule with checkLoops=false because it's annoying that it complains on something as idiomatic as `while (true)`

Reply:
FWIW my preference is for no eslint as I don't think it provides value beyond typescript + prettier, only friction.